### PR TITLE
Added release drafter for one click release of SDK

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
This PR adds a release drafter required for [1-click-release-process](https://github.com/opensearch-project/opensearch-build/blob/main/ONBOARDING.md#onboarding-to-universal--1-click-release-process). 

- [x] Tested the workflow [here](https://github.com/owaiskazi19/opensearch-sdk-java/actions/runs/5338941475/jobs/9677003118) and published a release on fork [here](https://github.com/owaiskazi19/opensearch-sdk-java/releases/tag/0.1).
- [x] Created a new task in gradle to fetch the version in release-drafter.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-sdk-java/issues/836

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshpalis — APPROVED** [review]

**Comment by owaiskazi19** [comment]
Let's wait for @gaiksaya to review before we merge this.

File: .github/workflows/release-drafter.yml
Comment (dbwiddis) [inline .github/workflows/release-drafter.yml:6]: From what I can see this will create a release issue for every push/merge to any branch.

Is that what's intended?  Seems we'll have a lot of issue spam, most of which will be "no" clicks.

This would create an issue on PR merge, plus when the backport workflow pushed to the backport branch it would create another identical issue for that temporary branch.   It might also create a release issue for every mend version bump PR.  I suspect we need some branch filtering at a minimum.

And should we not have a manual way to trigger this workflow with `on:` `workflow_dispatch:`

**Review by gaiksaya — APPROVED** [review]
LGTM! Will work on adding other elements of the release process.
Thanks!

File: .github/workflows/release-drafter.yml
Comment (gaiksaya) [reply .github/workflows/release-drafter.yml:6]: This will create a `draft release` on GitHub with the artifacts attached that need to be published.
More information on how the end to end process works: https://github.com/opensearch-project/opensearch-build/issues/1234

Comment (dbwiddis) [reply .github/workflows/release-drafter.yml:6]: Ah, got it, I forgot tags are a git thing and was thinking github labels!

**Review by dbwiddis — APPROVED** [review]

**Pull request merged**
